sys/cheri/cheric.h

```diff
@@ -113,6 +113,24 @@ cheri_is_address_inbounds(const void * __capability cap, vaddr_t addr)
 {
 	return (addr >= cheri_getbase(cap) && addr < cheri_gettop(cap));
 }
+
+/*
+ * Check whether a capability is NULL-derived
+ */
+#ifdef __cplusplus
+static __always_inline inline bool
+#else
+static __always_inline inline _Bool
+#endif
+cheri_is_null_derived(const void * __capability cap)
+{
+
+	return (cheri_gettag(cap) == 0 &&
+		cheri_getoffset(cap) == cheri_getaddress(cap) &&
+		cheri_getlen(cap) == -1UL &&
+		cheri_getperm(cap) == 0);
```

Comment:
This misses flags, reserved bits, etc
```suggestion
	return (__builtin_cheri_equal_exact((uintcap_t)cheri_getaddress(cap), cap));
```
There should probably be an `__builtin_cheri_is_null_derived()`

Reply:
printf() also has a check like this.  If nothing else having a 'cheri_is_null_derived()' in <sys/cheri/cheric.h> would be useful even in 'dev' for use in printf().